kerastuner/engine/hyperparameters.py

```diff
@@ -591,23 +596,48 @@ def Fixed(self,
                               parent_name=parent_name,
                               parent_values=parent_values)
 
+    @property
+    def space(self):
+        return list([hp for hp in self._space.values()])
+
     def get_config(self):
         return {
             'space': [{'class_name': p.__class__.__name__,
-                       'config': p.get_config()} for p in self.space],
+                       'config': p.get_config()}
+                      for p in self._space.values()],
             'values': dict((k, v) for (k, v) in self.values.items()),
         }
 
     @classmethod
     def from_config(cls, config):
         hp = cls()
-        hp.space = [deserialize(p) for p in config['space']]
+        for p in config['space']:
+            p = deserialize(p)
+            hp._space[p.name] = p
         hp.values = dict((k, v) for (k, v) in config['values'].items())
         return hp
 
     def copy(self):
         return HyperParameters.from_config(self.get_config())
 
+    def merge(self, hps, overwrite=True):
+        """Merges hyperparameters into this object.
+
+        Arguments:
+          hps: A `HyperParameters` object of list of `HyperParameter`
+            objects.
+          overwrite: bool. If `True`, the parameters in `hps` will
+            overwrite any values with the same name in this object.
+        """
+        frozen = self._frozen
+        self._frozen = not overwrite
```

Comment:
I would recommend removing `self._frozen` since:
- it isn't configurable
- it's only used in this method
- the name suggests that the state of the hps is frozen, when in fact it's always possible to add new entries

Instead, an `overwrite` argument in `_retrieve should be sufficient

Reply:
Done